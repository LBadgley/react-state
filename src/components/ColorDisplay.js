import React from 'react';
import PropTypes from 'prop-types';
import styles from './ColorPicker.css';

function ColorDisplay({ selectedColor }) {
  return <div className={styles.ColorDisplay} style={{ backgroundColor: selectedColor }}>{selectedColor}</div>;
}

ColorDisplay.propTypes = {
  selectedColor: PropTypes.string.isRequired
};

export default ColorDisplay;
