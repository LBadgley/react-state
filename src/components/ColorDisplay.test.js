import React from 'react';
import { shallow } from 'enzyme';
import ColorDisplay from './ColorDisplay';

describe('color display test', () => {
  it('displays the color', () => {
    const wrapper = shallow(<ColorDisplay color="blue"/>);
    expect(wrapper).toMatchSnapshot();
  });
});
