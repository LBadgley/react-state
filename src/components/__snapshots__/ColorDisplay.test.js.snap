// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`color display test displays the color 1`] = `
<div
  className="ColorDisplay"
  style={
    Object {
      "backgroundColor": "blue",
    }
  }
>
  blue
</div>
`;
